atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServices.java

```diff
@@ -279,13 +279,35 @@ static String getFilteredStackTrace(String filter) {
     }
 
     static Column createColumn(Cell cell, Value value) {
+        return createColumnAtSpecificCassandraTimestamp(cell, value, value.getTimestamp());
```

Comment:
I think this PR would be much, much simpler if we just wrote _everything_ in with the timestamp the supplier is currently giving us

Reply:
sorry, I know this is kinda the opposite of what we suggested before, but given the sweep sentinels codepath and the caching we're doing the cost should be minimal.

Basically always force the timestamp to the one the supplier's given us (assuming that the caching is cheap), and never have it be the one that is passed in by the user.